### PR TITLE
Add complexity benchmark

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR extends the benchmarking suite to also test the complexity scaling of the SCDB package by measuring runtime a function of dataset size.

The benchmark in its current form tests only the time it takes to create the initial snapshot (i.e. create an empty table)

Question for the reviewers: Do we want to extend the benchmarks further and time a update step as well?

### Approach
A second benchmark was added where tables of increasing size are written to a new database table.

The benchmark is added to describe the complexity analysis.

Example render of the new complexity figure:
billede

### Known issues
N/A

### Checklist

* [x] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [x] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

**Comment by marcusmunch** [comment]
I am not sure this is a useful way to report update times for a number of reasons.

First, `update_snapshot` has several potential time-consuming steps, each with potential benchmarking times per backend:

* Importing and computing input `.data`
* Computing and comparing checksums
* Identifying rows to insert/delete from a table, and doing this

The cost of these individual steps is greatly impacted by the number of rows existing/inserted/deactivated, and I only see much useful information in only reporting process times where these three values are the same. Also, I would expect the time spent by `update_snapshot` to go up quickly if `.data` is a local `data.frame`-like object 😇 

Finally, is it not fair to assume that duration will increase with number of rows affected? I would personally prefer the Y axis to show time in seconds per 1,000 rows. I assume this will reveal how much the number of rows affected impacts performance more clearly than having to evaluate the slope of each line.

**Comment by RasmusSkytte** [comment]
I am also not keen on benchmarking the update times - mostly because there are quite a number of ways to do it.
I may attempt it at some point, and then add it as a later PR if the results seem meaningful.

Regarding how to plot the complexity scaling, I went for the style I have seen the most (i.e. y-axis as total time, not time per row).
The goal, in my opinion, is to discern what type of scaling (e.g. linear, sub-linear or super-linear) and not to read the exact details from the graph.
(This is also the style on wikipedia: https://en.wikipedia.org/wiki/Big_O_notation#/media/File:Comparison_computational_complexity.svg)

If we plot the time per 1000-rows on the y-axis, it looks like this:
billede

Here we see that the efficiency of the program increases as the number of records increase (since we have some parts of the program that has a constant cost). It also looks like it asymptotically approaches a linear complexity (i.e. a flat line in this representation).

I personally prefer the original method for plotting, but I am open for input as to what the reviewers prefer.

File: vignettes/benchmarks.Rmd [hunk 1]
```diff
@@ -71,29 +74,43 @@ if (on_main) {
...
+slow_backends <- benchmark_1 |>
   dplyr::distinct(.data$database, .data$n) |>
   dplyr::filter(.data$n < max(.data$n)) |>
   dplyr::pull("database")
 
-benchmarks <- benchmarks |>
+benchmark_1 <- benchmark_1 |>
   dplyr::mutate("database" = paste0(database, ifelse(database %in% slow_backends, "*", "")))
 
+# Insert newline into databse name to improve rendering of figures
```
Comment (LasseEngboChr) [inline vignettes/benchmarks.Rmd:99]: databse -> database

File: vignettes/benchmarks.Rmd [hunk 2]
```diff
@@ -102,3 +119,39 @@ if (length(slow_backends) > 1) {
 
 g
 ```
+
+
+We include another benchmark to highlight the complexity scaling of the `update_snapshot() ` with the size of the input
+data. The datasets are similar to the first benchmark is used, but the number of repeats is varied to see the impact of
```
Comment (LasseEngboChr) [inline vignettes/benchmarks.Rmd:125]: delete: 'is used'

**Review by LasseEngboChr — APPROVED** [review]

**Review by LasseEngboChr — APPROVED** [review]

**Review by LasseEngboChr — APPROVED** [review]

**Pull request merged**